projects/vdk-core/src/vdk/internal/builtin_plugins/run/job_input.py

```diff
@@ -60,8 +61,8 @@ def __init__(
 
     # Connections
 
-    def get_managed_connection(self) -> ManagedConnectionBase:
-        return self.__managed_connection_builder.open_default_connection()
+    def get_managed_connection(self) -> PEP249Connection:
```

Comment:
What is the reason for changing the contract

Reply:
It was a mistake in the first place. I never intended for get_managed_connection()  to return ManagedConnectionBase.

Reply:
I may actually remember a discussion

Reply:
Possible. But get_managed_connection() is data engineering facing method and ManagedConnectionBase is mostly useful for plugin development to be subclass (though connection hooks should be preferred). Otherwise it's implementation detailed. Both execute_query and get_managed_connection (which ManagedConnectionBase offers) are exposed through IJobInput interface.

Reply:
I don't think mixing multiple changes to push a change using the gravity of another change, is optimal.
This should go into a separate PR.